src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -159,29 +168,25 @@ public void writeThrough(User user) {
     userEntity.setProperty("username", user.getName());
     userEntity.setProperty("password", user.getPassword());
     userEntity.setProperty("about", user.getAbout());
-    userEntity.setProperty("creation_time", user.getCreationTime().toString());
+    userEntity.setProperty("messagesSent", user.getMessagesSent());
+    userEntity.setProperty("allowMessageDel", user.getAllowMessageDel());
+    userEntity.setProperty("creation", user.getCreationTime().toString());
     datastore.put(userEntity);
   }
 
-  /** Change some property of a user then re-add to datastore. Currently only changing the
-   * the about message are supported. This method may be inefficient 
+  /** Change some property of a user then re-add to datastore. This method may be inefficient 
    * when there are many users. 
    */
   public void update(User user) {
-    /** This will be inefficient for many users. I believe this could be done in constant time without 
-     *  storing entity objects if the Entity objects were saved with known keys related to the users
-     *  (keys as usernames or user IDs), though I believe this would require remaking the users currently 
-     *  stored in the datastore in order to reassign the keys. 
-     *  --Abby 
-     */
-    for (Entity userEntity : userEntities) {
-      String userName = (String) userEntity.getProperty("username");
-      if (userName.equals(user.getName())) {
-        userEntity.setProperty("about", user.getAbout());
-        datastore.put(userEntity);
-        break;
-      }
-    }
+    UUID userId = user.getId();
+    if (!userEntitiesById.containsKey(userId))
+      return;
+    
```

Comment:
Always use semi colons around if statements. While this is correct, it lends clarity with semi colons.